src/test/http_openai_handler_test.cpp

```diff
@@ -243,3 +245,252 @@ TEST_F(HttpOpenAIHandlerTest, GraphWithANameDoesNotExist) {
+    EXPECT_EQ(apiHandler->parseMessages(), absl::OkStatus());
+    std::vector<ov::Tensor> images = apiHandler->getImages();
+    ASSERT_EQ(images.size(), 1);
+    ov::Tensor image = images[0];
+    EXPECT_EQ(image.get_element_type(), ov::element::u8);
+    EXPECT_EQ(image.get_size(), 3);
+    std::vector<uint8_t> expectedBytes = {160, 181, 110};
+    for (size_t i = 0; i < image.get_size(); i++) {
+        EXPECT_EQ(expectedBytes[i], ((uint8_t*)image.data())[i]);
+    }
+    json = apiHandler->getProcessedJson();
+    EXPECT_EQ(json, std::string("{\"model\":\"llama\",\"messages\":[{\"role\":\"user\",\"content\":\"What is in this image?\"}]}"));
+}
+
+TEST_F(HttpOpenAIHandlerParsingTest, ParsingMultipleMessages) {
+    std::string json = R"({
+    "model": "llama",
+    "messages": [
+      {
+        "role": "user",
+        "content": [
+          {
+            "type": "text",
+            "text": "What is in this image?"
+          },
+          {
+            "type": "image_url",
+            "image_url": {
+              "url":  "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAAEElEQVR4nGLK27oAEAAA//8DYAHGgEvy5AAAAABJRU5ErkJggg=="
+            }
+          }
+        ]
+      },
+      {
+        "role": "user",
+        "content": [
+          {
+            "type": "text",
+            "text": "What is in this image?"
+          },
+          {
+            "type": "image_url",
+            "image_url": {
+              "url":  "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAAEElEQVR4nGLK27oAEAAA//8DYAHGgEvy5AAAAABJRU5ErkJggg=="
+            }
+          }
+        ]
+      },
+      {
+        "role": "system",
+        "content": "What is Openvino?"
+      }
+    ]
+  })";
+    doc.Parse(json.c_str());
+    ASSERT_FALSE(doc.HasParseError());
+    std::shared_ptr<ovms::OpenAIChatCompletionsHandler> apiHandler = std::make_shared<ovms::OpenAIChatCompletionsHandler>(doc, ovms::Endpoint::CHAT_COMPLETIONS, std::chrono::system_clock::now(), *tokenizer);
+    EXPECT_EQ(apiHandler->parseMessages(), absl::OkStatus());
+    std::vector<ov::Tensor> images = apiHandler->getImages();
+    ASSERT_EQ(images.size(), 2);
+    std::vector<uint8_t> expectedBytes = {160, 181, 110};
+    for (auto image : images) {
+        EXPECT_EQ(image.get_element_type(), ov::element::u8);
+        EXPECT_EQ(image.get_size(), 3);
+        for (size_t i = 0; i < image.get_size(); i++) {
+            EXPECT_EQ(expectedBytes[i], ((uint8_t*)image.data())[i]);
+        }
+    }
+    json = apiHandler->getProcessedJson();
+    EXPECT_EQ(json, std::string("{\"model\":\"llama\",\"messages\":[{\"role\":\"user\",\"content\":\"What is in this image?\"},{\"role\":\"user\",\"content\":\"What is in this image?\"},{\"role\":\"system\",\"content\":\"What is Openvino?\"}]}"));
+}
+
+TEST_F(HttpOpenAIHandlerParsingTest, ParsingMessagesWithInvalidContentType) {
```

Comment:
In general, the new test names do not say what is expected behavior, my suggestion is that it could contain it, for example 
ParsingMultipleMessages -> ParsingMultipleTextMessagesWithImagesSucceeds
ParsingMessagesImageStringWithNoPrefix -> ParsingMessagesImageStringWithNoPrefixSucceeds

what do you think?